bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -156,7 +151,12 @@ public String toString() {
                 if (propertyValue == null) {
                     writer.writeNull();
                 } else {
-                    propertyModel.getCachedCodec().encode(writer, propertyValue, encoderContext);
+                    try {
+                        propertyModel.getCachedCodec().encode(writer, propertyValue, encoderContext);
+                    } catch (CodecConfigurationException e) {
```

Comment:
Encoding exceptions now match decoding exceptions - because the change now lazily throws an exception.